src/core/qgsvectorlayer.cpp

```diff
@@ -2387,6 +2387,19 @@ long QgsVectorLayer::featureCount() const
          ( mEditBuffer ? mEditBuffer->mAddedFeatures.size() - mEditBuffer->mDeletedFeatureIds.size() : 0 );
 }
 
+long QgsVectorLayer::featureCount( QgsFeatureRequest request )
```

Comment:
It is probably useful. But I like classes with no more than needed :) QgsVectorLayer is already overcrowded. I would see it in something like a size_t QgsVectorLayerUtils::featureCount( const QgsVectorLayer&, QgsFeatureRequest )
or QgsFeatureRequestUtils::featureCount( ...)